src/newscale/stages.py

```diff
@@ -1,365 +1,497 @@
+            self.stages[axis_name].run(direction, seconds)
+        if not wait:
+            return
+        # Poll position vector until we have reached the target or timeout.
+        start_time = perf_counter()
+        while perf_counter() - start_time < self.__class__.MOVE_TIMEOUT_S:
+            stats = {x: self.stages[x].get_closed_loop_state_and_position()[0]
+                     for x in axes.keys()}
+            if any([stats[x][StateBit.STALLED] for x in axes.keys()]):
+                raise RuntimeError("One or more axes is stalled.")
+            if all([(not stats[x][StateBit.RUNNING])
+                    and (not stats[x][StateBit.TIMED_RUN])
+                    for x in axes.keys()]):
+                return
+            sleep(0.01)
+        raise RuntimeError("One or more axes timed out trying to move for the"
+                           "specified time.")
+
+    @axis_check()
+    def get_position(self, *axes: str):
+        """Retrieve the specified axes positions (in mm) as a dict, or get all
+        positions if none are specified.
+
+        :param axes: an unlimited number of axes specified by name (string).
+
+        .. code-block:: python
+
+            stages.get_position('x', 'y', 'z')  # Get specified positions OR
+            stages.get_position()  # Get all positions.
+
+        """
+        if len(axes) == 0:  # return all axes if None are specified.
+            axes = self.stages.keys()
+        return {x: self.stages[x].get_position() for x in axes}
+
+    @axis_check()
+    def set_speed(self, **axes):
+        """Set the speeds of the specified axes in mm/sec."""
+        # TODO: Units
+        pass
+
+    def get_speed(self, **axes):
+        """Get the speeds of the specified axes"""
+        # TODO: Units
+        pass
+
+    @axis_check()
+    def halt(self):
+        """Halt all axes"""
+        return {x: self.stages[x].halt() for x in self.stages.keys()}
 
+    def close(self):
+        """Release computer control of all axes."""
+        return {x: self.stages[x].close() for x in self.stages.keys()}
+
+
+class USBXYZStage(MultiStage):
+    """An XYZ Stage from a single USB interface."""
+
+    def __init__(self, port: str = None,
+                 serial_interface: SerialInterface = None):
+        if not ((port is None) ^ (serial_interface is None)):
+            raise SyntaxError("Exclusively either port or serial_interface"
+                              "(i.e: one or the other, but not both) options "
+                              "must be specified.")
+        self.interface = serial_interface if serial_interface and not port \
+            else SerialInterface(port)
+        # Create 3 stages with corresponding addresses.
+        stages = {'x': M3LinearSmartStage(self.interface, '01'),
+                  'y': M3LinearSmartStage(self.interface, '02'),
+                  'z': M3LinearSmartStage(self.interface, '03')}
+        super().__init__(**stages)
+
+
+class PoEXYZStage(MultiStage):
```

Comment:
let's get rid of PoE stuff for this PR

Reply:
Same comment from:
https://github.com/AllenNeuralDynamics/python-newscale/pull/1#discussion_r1001107903

Reply:
same as above, let's do it in a separate PR

Reply:
Done!